include/circt/Dialect/Seq/Seq.td

```diff
@@ -47,10 +50,31 @@ def CompRegOp : SeqOp<"compreg",
     Optional<I1>:$reset, Optional<AnyType>:$resetValue);
   let results = (outs AnyType:$data);
 
-  let printer = "return ::print(p, *this);";
+  let printer = "return ::print$cppClass(p, *this);";
   let parser = "return ::parse$cppClass(parser, result);";
 }
 
+def RegOp : SeqOp<"reg",
+    [AllTypesMatch<["input", "data"]>, SameVariadicOperandSize]> {
+
+  let summary = "A complete implementation of registers.";
+  let description = "See the Seq dialect rationale for a longer description";
+
+  let arguments = (ins AnyType:$input, I1:$clk,
+    I1:$enable,
+    Optional<I1>:$reset, Optional<AnyType>:$resetValue,
```

Comment:
What's the rational for reset being optional but enable not being optional?  Always enable is enable tied to 1, never reset if reset tied to 0.